Libplanet/Action/IActionEvaluation.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using Bencodex.Types;
+
+namespace Libplanet.Action
+{
+    public interface IActionEvaluation
+    {
+        /// <summary>
+        /// An action data to evaluate.
+        /// </summary>
+        public IValue? Action { get; }
```

Comment:
Why is this nullable? Do we need an `ActionEvaluation` without any `IAction`? 🤨

Reply:
I missed the documentation. When the `Action` is `null`, it means block action (i.e., `IBlockPolicy.BlockAction`). because block action cannot be deserialized with `PolymorphicAction<T>` and I think block action shouldn't be serialized and deserialized.

I'll complement the documentation for when the `Action` is `null`.

Reply:
I updated the documentation like the below:

```csharp
        /// <summary>
        /// An action data to evaluate. When the value is null, it means this action evaluation
        /// evaluated the <see cref="IBlockPolicy{T}.BlockAction"/>.
        /// </summary>
```

Reply:
Now I understand why. I think it could be a source of confusion though.

> The main key to this pull request is to change the type of the IActionEvaluation.Action to IValue? instead of IAction and to pass the responsibility of interpreting this IActionEvaluation.Action to the receiving end.

Why is this necessary again? My understanding is that for an action to be evaluated, it should've been concretized via an evaluator. Will it be changed back to `IAction` again? 🤔

Reply:
> Why is this necessary again? My understanding is that for an action to be evaluated, it should've been concretized via an evaluator. Will it be changed back to `IAction` again? 🤔

(I'm not sure I understood your question.) As you said, to evaluate an action, it should be concretized via an evaluator. It will not be changed back to `IAction`.